lib/common/compiler.h

```diff
@@ -91,18 +91,29 @@
 /* prefetch
  * can be disabled, by declaring NO_PREFETCH macro */
 #if defined(NO_PREFETCH)
-#  define PREFETCH(ptr)   /* disabled */
+#  define PREFETCH(ptr)     (void)(ptr)  /* disabled */
 #else
 #  if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_I86))  /* _mm_prefetch() is not defined outside of x86/x64 */
 #    include <mmintrin.h>   /* https://msdn.microsoft.com/fr-fr/library/84szxsww(v=vs.90).aspx */
-#    define PREFETCH(ptr)   _mm_prefetch((const char*)ptr, _MM_HINT_T0)
+#    define PREFETCH(ptr)   _mm_prefetch((const char*)(ptr), _MM_HINT_T1)
 #  elif defined(__GNUC__) && ( (__GNUC__ >= 4) || ( (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1) ) )
-#    define PREFETCH(ptr)   __builtin_prefetch(ptr, 0, 0)
+#    define PREFETCH(ptr)   __builtin_prefetch((ptr), 0 /* rw==read */, 2 /* locality */)
 #  else
-#    define PREFETCH(ptr)   /* disabled */
+#    define PREFETCH(ptr)   (void)(ptr)  /* disabled */
 #  endif
 #endif  /* NO_PREFETCH */
 
+#define CACHELINE_SIZE 64
+
+#define PREFETCH_AREA(p, s)  {            \
```

Comment:
arg, creating an inline function without triggering dreaded warnings when function defined in the `.h` is not invoked in the `.c` does require existing macro `MEM_STATIC`, which is [defined in `mem.h`](https://github.com/facebook/zstd/blob/dev/lib/common/mem.h#L26), but is not a dependency of `compiler.h`.

Several solutions : 
- make `mem.h` a dependency of `compiler.h` => feels overkill
- duplicate `MEM_STATIC` definition (and eventually rename it) within `compiler.h`
- move `MEM_STATIC` out of `mem.h`, where it's strange anyway (was there for its own purpose initially), and declare it instead within `compiler.h`, then make `compiler.h` a dependency of `mem.h`. Optionally rename the macro since it's no longer in `mem.h` (though we have a ton of code depending on this macro, that will need to be updated) => too large change, dwarfing the current patch.
- don't change anything, keep a macro.

Something like _3_ seems cleaner in the longer term, but I can't help feeling this is overkill for the current patch, and we are better off in short term to stick to _4_.